src/crypto/crypto_ec.cc

```diff
@@ -270,13 +278,19 @@ void ECDH::GetPrivateKey(const FunctionCallbackInfo<Value>& args) {
     return THROW_ERR_CRYPTO_OPERATION_FAILED(env,
         "Failed to get ECDH private key");
 
-  const int size = BN_num_bytes(b);
-  AllocatedBuffer out = AllocatedBuffer::AllocateManaged(env, size);
-  CHECK_EQ(size, BN_bn2binpad(b,
-                              reinterpret_cast<unsigned char*>(out.data()),
-                              size));
+  std::unique_ptr<BackingStore> bs;
+  {
+    NoArrayBufferZeroFillScope no_zero_fill_scope(env->isolate_data());
+    bs = ArrayBuffer::NewBackingStore(env->isolate(), BN_num_bytes(b));
+  }
+  CHECK_EQ(static_cast<int>(bs->ByteLength()),
+           BN_bn2binpad(
+               b, static_cast<unsigned char*>(bs->Data()), bs->ByteLength()));
 
-  args.GetReturnValue().Set(out.ToBuffer().FromMaybe(Local<Value>()));
+  Local<ArrayBuffer> ab = ArrayBuffer::New(env->isolate(), std::move(bs));
+  Local<Value> buffer;
+  if (!Buffer::New(env, ab, 0, ab->ByteLength()).ToLocal(&buffer)) return;
+  args.GetReturnValue().Set(buffer);
```

Comment:
nit: why not if the other way around and have one less (possibly hidden) `return`?
```suggestion
  if (Buffer::New(env, ab, 0, ab->ByteLength()).ToLocal(&buffer)) {
   args.GetReturnValue().Set(buffer);
  }
```